aleph2_logging_service/src/com/ikanow/aleph2/logging/service/LoggingService.java

```diff
@@ -0,0 +1,191 @@
+	
+	private final static Logger _logger = LogManager.getLogger();
+	protected final static Cache<String, MultiDataService> bucket_writable_cache = CacheBuilder.newBuilder().expireAfterAccess(30, TimeUnit.MINUTES).build();
+	
+	protected final LoggingServiceConfigBean properties;
+	protected final IServiceContext service_context;
+	protected final IStorageService storage_service;
+	
+	@Inject
+	public LoggingService(
+			final LoggingServiceConfigBean properties, 
+			final IServiceContext service_context) {
+		this.properties = properties;
+		this.service_context = service_context;
+		this.storage_service = service_context.getStorageService();		
+	}	
+
+	/* (non-Javadoc)
+	 * @see com.ikanow.aleph2.data_model.interfaces.shared_services.ILoggingService#getLogger(com.ikanow.aleph2.data_model.objects.data_import.DataBucketBean)
+	 */
+	@Override
+	public CompletableFuture<IBucketLogger> getLogger(DataBucketBean bucket) {
+		return getBucketLogger(bucket, getWritable(bucket), false);		
+	}
+
+	/* (non-Javadoc)
+	 * @see com.ikanow.aleph2.data_model.interfaces.shared_services.ILoggingService#getSystemLogger(java.util.Optional)
+	 */
+	@Override
+	public CompletableFuture<IBucketLogger> getSystemLogger(DataBucketBean bucket) {
+		return getBucketLogger(bucket, getWritable(bucket), true);
+	}
+	
+	/* (non-Javadoc)
+	 * @see com.ikanow.aleph2.data_model.interfaces.shared_services.ILoggingService#getExternalLogger(java.lang.String)
+	 */
+	@Override
+	public CompletableFuture<IBucketLogger> getExternalLogger(final String subsystem) {
+		final DataBucketBean bucket = LoggingUtils.getExternalBucket(subsystem, Optional.ofNullable(properties.default_system_log_level()).orElse(Level.OFF));		
+		return getBucketLogger(bucket, getWritable(bucket), true);
+	}
+	
+	/**
+	 * Retrieves a writable for the given bucket, trys to find it in the cache first, creates a new one if it can't and adds it to the cache for future requests.
+	 * 
+	 * @param log_bucket
+	 * @return
+	 * @throws ExecutionException 
+	 */
+	private MultiDataService getWritable(final DataBucketBean log_bucket) {
+		try {
+			return bucket_writable_cache.get(getWritableCacheKey(log_bucket), () -> {
+				return LoggingUtils.getLoggingServiceForBucket(service_context, log_bucket);
+			});
+		} catch (ExecutionException e) {
+			//return an empty multiwriter when we've had a failure (this shouldn't occur, but justu to be safe)
+			_logger.error("Error getting writable for bucket: " + log_bucket.full_name() + " return an empty logger instead that ignores requests", e);
+			return LoggingUtils.getLoggingServiceForBucket(service_context, LoggingUtils.getEmptyBucket());
+		}
+	}	
+
+	/**
+	 * Creates the bucket logger for the given bucket.  First attempts to write the
+	 * output path, if that fails returns an exceptioned completable.
+	 * 
+	 * @param bucket
+	 * @param writable
+	 * @param b
+	 */
+	private CompletableFuture<IBucketLogger> getBucketLogger(DataBucketBean bucket,
```

Comment:
Don't think these getBucketLogger calls should return CFs as discussed, for consistency with everthing else